### PR TITLE
[WIP]Hawaii pupa scraper issue #1442

### PR DESCRIPTION
**Pull request by jamesthechamp**
hawaii scraper conversion to pupa framework

- [x] people 

- [x] events

- [x] votes & bills

- [x] committees

### CODE REVIEW

**Comment by jamesthechamp** [comment]
#1442 
Hawaii pupa conversion completed
one issue left 
committees related to bill action are not stored.

File: billy_metadata/hi.py [hunk 1]
```diff
@@ -0,0 +1,119 @@
+# from billy.utils.fulltext import pdfdata_to_text, text_after_line_numbers
```
Comment (jmcarp) [inline billy_metadata/hi.py]: Can these commented-out imports be deleted?

File: billy_metadata/hi.py [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+# from .bills import HIBillScraper
+# from .legislators import HILegislatorScraper
+# from .events import HIEventScraper
+
+settings = dict(SCRAPELIB_TIMEOUT=300)
```
Comment (jmcarp) [inline billy_metadata/hi.py]: I don't think this line does anything here, but it should probably be restored in `openstates/hi/__init__.py`.

File: billy_metadata/hi.py [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+def session_list():
+    # doesn't include current session, we need to change it
+    from billy.scrape.utils import url_xpath
+    sessions = url_xpath('http://www.capitol.hawaii.gov/archives/main.aspx',
+            "//div[@class='roundedrect gradientgray shadow']/a/text()"
+        )
+    sessions.remove("Archives Main")
+    return sessions
+
+def extract_text(doc, data):
```
Comment (jmcarp) [inline billy_metadata/hi.py]: Is this function used?

File: billy_metadata/hi.py [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+    _ignored_scraped_sessions = [
+        # ignore odd years after they're over..
+        '2011',
+        '2010', '2009', '2008', '2007', '2006',
+        '2005', '2004', '2003', '2002', '2001',
+        '2000', '1999'
+    ]
+)
+
+def session_list():
```
Comment (jmcarp) [inline billy_metadata/hi.py]: I think this can be moved to a `get_session_list` method of the `Hawaii` class.

File: openstates/hi/bills.py [hunk 5]
```diff
@@ -117,28 +114,40 @@ def parse_bill_actions_table(self, bill, action_table):
...
                     except KeyError:
                         pass
-
-            bill.add_action(actor, string, date,
-                            type=act_type, committees=real_committees)
-
+            act = bill.add_action(string, date, chamber=actor,
+                                  classification=act_type)
+            for committee in real_committees:
+                print(committee, bill_id)
```
Comment (jmcarp) [inline openstates/hi/bills.py]: Can this be deleted, or is it useful to keep?

File: openstates/hi/bills.py [hunk 6]
```diff
@@ -117,28 +114,40 @@ def parse_bill_actions_table(self, bill, action_table):
...
+                                 legislative_session=session,
+                                 motion_text=motion,
+                                 result='pass' if 'passed' in string.lower() else 'fail',
+                                 classification='passage')
+                vote.add_source(url)
+                vote.set_count('yes', int(v['n_yes'] or 0))
+                vote.set_count('no', int(v['n_no'] or 0))
+                vote.set_count('not voting', int(v['n_excused'] or 0))
+                for voter in split_specific_votes(v['yes']):
+                    # getattr(vote, attrib)(voter)
```
Comment (jmcarp) [inline openstates/hi/bills.py]: I think we can delete these commented lines.

File: openstates/hi/bills.py [hunk 7]
```diff
@@ -156,62 +165,66 @@ def parse_bill_versions_table(self, bill, versions):
...
             subs.remove("")
 
-        b = Bill(session, chamber, bill_id, title=meta['Measure Title'],
-                 summary=meta['Description'],
-                 referral=meta['Current Referral'],
-                 subjects=subs,
-                 type=bill_type)
-        b.add_source(url)
-
+        # b = Bill(session, , bill_id, title=meta['Measure Title'],
```
Comment (jmcarp) [inline openstates/hi/bills.py]: These can be deleted too.

File: openstates/hi/events.py [hunk 8]
```diff
@@ -28,70 +22,74 @@ def get_related_bills(self, href):
...
                     bill['bill_id'],
-                    description=bill['descr'],
-                    type=bill['type']
+                    note=bill['type']
                 )
 
-            self.save_event(event)
+            yield event
+
+    def scrape(self):
```
Comment (jmcarp) [inline openstates/hi/events.py]: Since this method just calls `scrape_events`, could we drop `scrape_events` and move that code here?

**Comment by jmcarp** [comment]
This looks good! Could you check for print statements and commented code that can be deleted?

File: openstates/hi/people.py [hunk 9]
```diff
@@ -0,0 +1,241 @@
...
+            fName = els[2].text_content()
+            return "%s %s" % (fName, lName)
+
+        def _scrape_party(els):
+            party = {
+                "(D)": "Democratic",
+                "(R)": "Republican"
+            }
+
+            try:
```
Comment (jmcarp) [inline openstates/hi/people.py]: Looks like this block can be simplified to `return party.get(els[4].text_content(), 'Other')`.

File: openstates/hi/people.py [hunk 10]
```diff
@@ -0,0 +1,241 @@
...
+
+    def scrape_contact_info(self, contact):
+        homepage = "%s/%s" % (
+            HI_BASE_URL,
+            contact.xpath("./a")[0].attrib['href']
+        )
+
+        els = self.br_split(contact)
+
+        def _scrape_title(els):
```
Comment (jmcarp) [inline openstates/hi/people.py]: I see that the code looked this way before this pull request, but do you know why these functions are defined inside a method and not at the class or module level?

**Comment by jmcarp** [comment]
Also, since pupa doesn't automatically create committees when we add memberships to a person (h/t @jamesturk, and correct me if I'm wrong), we'll need to explicitly yield `Organization` objects somewhere so that we continue to save committees correctly.

File: billy_metadata/hi.py (hunk 1 above)
Comment (jamesthechamp) [inline billy_metadata/hi.py]: done

File: billy_metadata/hi.py (hunk 2 above)
Comment (jamesthechamp) [inline billy_metadata/hi.py]: i kept it for now .. but pupa is not updating this is setting...

File: billy_metadata/hi.py (hunk 3 above)
Comment (jamesthechamp) [inline billy_metadata/hi.py]: not really so ... removed

File: billy_metadata/hi.py (hunk 4 above)
Comment (jamesthechamp) [inline billy_metadata/hi.py]: done

File: openstates/hi/bills.py (hunk 5 above)
Comment (jamesthechamp) [inline openstates/hi/bills.py]: sorry for that i just removed all such used print

File: openstates/hi/bills.py (hunk 6 above)
Comment (jamesthechamp) [inline openstates/hi/bills.py]: done

File: openstates/hi/bills.py (hunk 7 above)
Comment (jamesthechamp) [inline openstates/hi/bills.py]: done

File: openstates/hi/people.py (hunk 9 above)
Comment (jamesthechamp) [inline openstates/hi/people.py]: ya it can be .. and changed

File: openstates/hi/people.py (hunk 10 above)
Comment (jamesthechamp) [inline openstates/hi/people.py]: their is no good reason why cant we put this function to class level... so i just changed it so...

File: openstates/hi/events.py (hunk 8 above)
Comment (jamesthechamp) [inline openstates/hi/events.py]: sure

**Comment by jamesthechamp** [comment]
@jmcarp i am gonna make a committee scraper for above issue you mention....

**Comment by jamesthechamp** [comment]
all done

**Pull request merged**